argus/backend/service/results_service.py

```diff
@@ -365,12 +368,25 @@ def _get_tables_metadata(self, test_id: UUID) -> list[ArgusGenericResultMetadata
         tables_meta = self.cluster.session.execute(query=query, parameters=(test_id,))
         return [ArgusGenericResultMetadata(**table) for table in tables_meta]
 
-    def _get_tables_data(self, test_id: UUID, table_name: str, ignored_runs: list[RunId]) -> list[ArgusGenericResultData]:
+    def _get_tables_data(self, test_id: UUID, table_name: str, ignored_runs: list[RunId],
+                         start_date: datetime | None = None, end_date: datetime | None = None) -> list[ArgusGenericResultData]:
         query_fields = ["run_id", "column", "row", "value", "status", "sut_timestamp"]
         raw_query = (f"SELECT {','.join(query_fields)}"
-                     f" FROM generic_result_data_v1 WHERE test_id = ? AND name = ? LIMIT 2147483647")
+                     f" FROM generic_result_data_v1 WHERE test_id = ? AND name = ?")
+
+        parameters = [test_id, table_name]
+
+        if start_date:
+            raw_query += " AND sut_timestamp >= ?"
+            parameters.append(start_date)
+        if end_date:
+            raw_query += " AND sut_timestamp <= ?"
+            parameters.append(end_date)
+
+        if start_date or end_date:
+            raw_query += " ALLOW FILTERING"
```

Comment:
do we have some estimation of the performance of those queries ? before/after ?

Reply:
No, but in general should be faster as there will be less data sent to the backend.
Possibly this whole table could be improved by making SUT_TIMESTAMP a static field - as we don't support different SUT versions for one test run. I'd consider doing this in separate PR. WDYT?